Use bearer token instead of cookie for auth

majakomel opened the pull request.

hellais commented on components/lib/api.js:
```diff
@@ -77,6 +90,9 @@ export const registerUser = async (email_address, redirect_to) => {
 
 export const loginUser = async (token) => {
   return await getAPI(apiEndpoints.USER_LOGIN, { k: token })
+    .then((response) => {
```
I suspect that since this is going through the `getAPI` call if the user decides to pre-emptively login again (getting a new token), while the old one currently in the browser is expired, they will end up with an error page instead of properly logging in.

Perhaps in the case of `login` we should bypass the `refreshToken` call since we are going to be getting a new token anyways.

hellais commented on components/lib/api.js:
```diff
@@ -10,51 +11,63 @@ export const apiEndpoints = {
-  withCredentials: true
 })
 
+const refreshToken = async () => {
```
I'm not sure if this is already done as a side-effect of the account_metadata call, but it might be worth explicitly having a periodic call to refreshToken so that we pre-emptively refresh it even if the user is not performing any actions on the page.

majakomel commented:
@hellais I have refactored the following:

- replaced `getAPI` with basic `axios.get` to avoid ambiugity and have a "clean" login request
- handling of 401 happens within `useUser` hook and takes use of existing logic for such cases
- `refreshToken` is placed within `setInterval` and is periodically checking if conditions for refresh are met (timings can be adjusted)

Changes are also available for testing on https://test-lists.test.ooni.org/

hellais commented on components/lib/api.js:
```diff
@@ -10,21 +11,29 @@ export const apiEndpoints = {
   SUBMISSION_SUBMIT: '/api/v1/url-submission/submit'
 }
 
+const token = () => {
```
Perhaps to prevent confusion with the `token` variable inside of `loginUser` this could be called `bearerToken` or `getBearerToken`?

hellais commented:
I left a small nitpick comment about changing a variable name. For the rest the changes look good and it can be merged.
